Firestore/Source/Model/FSTDocumentKey.h

```diff
@@ -18,10 +18,23 @@
 
 #include <initializer_list>
 
+#include "Firestore/core/src/firebase/firestore/model/document_key.h"
 #include "Firestore/core/src/firebase/firestore/model/resource_path.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
+// Somehow, when ObjC header includes a C++ that includes that ObjC header, the C++ class
+// declaration is not visible. So manually forward-declares the class here.
```

Comment:
I don't think that's the problem. There's an include loop here; this file includes document_key.h which includes this file.

The traditional way to resolve co-dependent header files is do create a forward declaration (like you've done.) A better way, would be to make them not co-dependent (but that's not always possible. Given the nature of this change, I suspect this to be the case here.)

So I'd recommend one of the following (in preferred order):
a) trying to make them not co-dependent (probably not possible), or
b) adjusting this comment.

Reply:
done